hack/build-coredns.sh

```diff
@@ -20,9 +28,13 @@ COREDNS_URL="https://github.com/openshift/coredns"
 if  [ ! -z "${1-}" ] && [ "${1}" = "upstream" ]
 then
     COREDNS_URL="https://github.com/coredns/coredns"
+    BRANCH_TAG="v1.11.3"
```

Comment:
Why this version is fixed? It's likely to change for subsequent releases (4.17+). Even for `release-4.16` it's not `1.11.3` ([the rebase PR](https://github.com/openshift/coredns/pull/118) is still not merged on master.)

Reply:
Do you have any suggestions of how to automate the version for upstream coredns rather than hardcoding?
For 4.16 the version used in 1.11.1 which I have used in the manual backport PR (#14).
For 4.17 I have used 1.11.3 as the story linked to the rebase PR belongs to an epic marked for 4.17.

Reply:
> Do you have any suggestions of how to automate the version for upstream coredns rather than hardcoding?

To automate: we can get the coredns version from [CoreVersion variable](https://github.com/openshift/coredns/blob/master/coremain/version.go#L5) of the branch we are in. But this may be a too much work for too little benefit.
For starters,  I propose to do a simple set of switch cases: if `release-4.16` then ` BRANCH_TAG=v1.11.1` unless it's the default (`v1.11.3`). New `if`s will be added with new releases.

Reply:
> For starters, I propose to do a simple set of switch cases: if `release-4.16` then ` BRANCH_TAG=v1.11.1` unless it's the default (`v1.11.3`). New `if`s can be added with new releases.

Aren't we doing the same thing now without the `if`s? If we add the `if`, it'll just keep on growing with newer releases. We can just change the version in the future releases whenever downstream coredns rebases to a new upstream version.

Reply:
It is similar to tagging the image in `.ci-operator.yaml` for each release branch: https://github.com/openshift/coredns-ocp-dnsnameresolver/blob/main/.ci-operator.yaml#L4

Reply:
The hardcoded tag is only for the upstream one. The downstream can still be deduced from the current branch name.

I initially thought that by deducing the downstream version, we can also arrive at the upstream tag. But then when I understood that using `if` statement won't solve the problem, it became evident that hardcoding the tag for the upstream coredns will be the easiest way forward.

I have hardocoded 1.11.1 in the backport PR (https://github.com/openshift/coredns-ocp-dnsnameresolver/pull/14).

Reply:
A simple `curl` from the downstream fork can deduce the upstream version:
```
$ BRANCH_TAG=release-4.16
$ curl -s https://raw.githubusercontent.com/openshift/coredns/${BRANCH_TAG}/coremain/version.go | grep CoreVersion | grep -Po '\d+\.\d+\.\d+'
1.11.1
```

Reply:
Made the required changes